roomacoustics/sti.py

```diff
@@ -0,0 +1,176 @@
+        1/2 RT60. [1],  p. ??
+
+    data_type : 'electrical', 'acoustical'
+        Determines weather input signals are obtained acoustically or electrically.
+        Auditory effects can only be considered when "acoustical" [1], p. ??. 
+        Default is 'none'.
+
+    gender: 'female', 'male'
+        Defines the applied weighting factors. Default is 'male' as the STI 
+        algorithm is initially based on the male speech spectrum.
+
+    noise: np.array, 0
+        Level of the test signal in octave for bands 125 Hz - 8000 Hz in dB_SPL.
+
+    snr: np.array, 0
+        SNR in octave for bands 125 Hz - 8000 Hz in dB.
+
+    """
+    
+
+# process signal, check input
+    sig = signal.copy()
+    #2D Signal
+    sig = sig.flatten()
+
+#check for sufficient signal length (IEC 60268-16 p. ## and p. ##) 
+    if signal.n_samples/sig.sampling_rate < 1.6:
+        warnings.warn("Signal length below 1.6 seconds.")
+
+# apply octave band filters (preliminary with crossover; later: perf. 
+# reconstructing oct. filter)
+    sig_oct = (filt.crossover(sig, 4, [90, 180, 355, 710, 1400, 2800, 5600,
+    11200])[1:-1,:])
+# MTF per octave and modulation frequency
+    mf = [0.63, 0.80, 1, 1.25, 1.60, 2, 2.5, 3.15, 4, 5, 6.3, 8, 10, 12]
+    mtf = np.zeros((14,)+sig_oct.cshape)
+    sig_en = np.sum(np.square(sig_oct.time),axis=-1)
+    sig_oct_newlength = sig_oct.copy()
+
+    #calculate mtf
+    with np.errstate(divide='ignore'):  #return nan for empty IR
+        for i in range(len(mf)):
+            fbin = int(np.ceil(mf[i]*sig_oct.n_samples/sig_oct.sampling_rate))
+            newlength = int(sig_oct.sampling_rate*fbin/mf[i])
+            sig_oct_newlength.time = np.concatenate((sig_oct.time, 
+                np.zeros((sig_oct.cshape + (newlength-sig_oct.n_samples,)))),
+                axis=-1)
+            sig_oct_newlength_sq = Signal(np.square(sig_oct_newlength.time),
+                sig_oct.sampling_rate)
+            mtf[i]=abs(sig_oct_newlength_sq.freq[...,fbin])/sig_en
+
+
+# Adjustment of mtf for ambient noise, auditory masking and threshold effects
+
+    # calculation only if levels are given
+    if not np.isnan(level).all():
+        # warn if SNR is below 20 dB in any band
+        level = level
+        snr = snr
+        if np.any(snr<20):
+            warnings.warn("SNR should be at least 20 dB for every octave band.")
+        # set snr to infinity if not given
+        if np.isnan(snr).all():
+            warnings.warn("SNR is not or incompletely given, calculation with \
+            infinite SNR.")
+            snr = np.ones(level.shape)*np.inf
+        # overall intensity
+        i_k = 10**(level/10)+10**((level-snr)/10)
+        # Apply ambient noise effects
+        mtf =  mtf*(10**(level.T/10)/i_k.T)
+
+        # consideration of auditory effects only for acoustical signals
+        if data_type == "electrical":
+            pass
+        else: 
+            if data_type != "acoustical":
```

Comment:
I would add a separate check at the beginning of the function and raise an error:
```
if data_type not in ["electrical", "acoustical"]:
    raise ValueError(f"data_type is {data_type} but must be 'electrical' or 'acoustical'.")
```